internal/services/nginx/nginx_configuration_resource.go

```diff
@@ -0,0 +1,318 @@
+			// todo remove set retry to 1
+			client.Client.RetryAttempts = 1
+			existing, err := client.ConfigurationsGet(ctx, id)
+			if !response.WasNotFound(existing.HttpResponse) && !response.WasStatusCode(existing.HttpResponse, http.StatusInternalServerError) {
+				if err != nil {
+					return fmt.Errorf("retreiving %s: %v", id, err)
+				}
+				return meta.ResourceRequiresImport(m.ResourceType(), id)
+			}
+
+			req := model.ToSDKModel()
+
+			future, err := client.ConfigurationsCreateOrUpdate(ctx, id, req)
+			if err != nil {
+				return fmt.Errorf("creating %s: %v", id, err)
+			}
+
+			if err := future.Poller.PollUntilDone(); err != nil {
+				return fmt.Errorf("waiting for creation of %s: %v", id, err)
+			}
+
+			meta.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (m ConfigurationResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, meta sdk.ResourceMetaData) error {
+			id, err := nginxconfiguration.ParseConfigurationID(meta.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			client := meta.Client.Nginx.NginxConfiguration
+			result, err := client.ConfigurationsGet(ctx, *id)
+			if err != nil {
+				return err
+			}
+
+			if result.Model == nil {
+				return fmt.Errorf("retrieving %s got nil model", id)
+			}
+
+			var output ConfigurationModel
+			output.Name = pointer.ToString(result.Model.Name)
+			deployID := nginxdeployment.NewNginxDeploymentID(id.SubscriptionId, id.ResourceGroupName, id.DeploymentName)
+			output.NginxDeploymentId = deployID.ID()
+
+			if prop := result.Model.Properties; prop != nil {
+				output.RootFile = pointer.ToString(prop.RootFile)
+
+				if prop.Package != nil && prop.Package.Data != nil {
+					output.PackageData = pointer.ToString(prop.Package.Data)
+				}
+
+				if files := prop.Files; files != nil {
+					for _, file := range *files {
+						output.ConfigFile = append(output.ConfigFile, ConfigFile{
+							Content:     pointer.ToString(file.Content),
+							VirtualPath: pointer.ToString(file.VirtualPath),
+						})
+					}
+				}
+
+				if files := prop.ProtectedFiles; files != nil {
+					for _, file := range *files {
+						output.ProtectedFile = append(output.ProtectedFile, ProtectedFile{
+							Content:     pointer.ToString(file.Content),
+							VirtualPath: pointer.ToString(file.VirtualPath),
+						})
+					}
+				}
+			}
+
+			return meta.Encode(&output)
+		},
+	}
+}
+
+func (m ConfigurationResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 10 * time.Minute,
+		Func: func(ctx context.Context, meta sdk.ResourceMetaData) (err error) {
+			client := meta.Client.Nginx.NginxConfiguration
+			id, err := nginxconfiguration.ParseConfigurationID(meta.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var model ConfigurationModel
+			if err = meta.Decode(&model); err != nil {
+				return fmt.Errorf("decoding err: %+v", err)
+			}
+
+			upd := model.ToSDKModel()
```

Comment:
(as above) we should inline this here and make use of `metadata.ResourceData.HasChanges("field")` as appropriate